worlds/witness/utils.py

```diff
@@ -5,6 +5,41 @@
 from fractions import Fraction
 
 
+def make_warning_string(any_j : bool, any_u : bool, any_d : bool, all_j : bool, all_u : bool, all_d : bool ):
```

Comment:
This function signature is breaking me, but i don't have a better suggestion. White spaces could look nicer though
```suggestion
def make_warning_string(any_j: bool, any_u: bool, any_d: bool, all_j: bool, all_u: bool, all_d: bool) -> str:
```